### PR TITLE
HDDS-6453. Ozone start/stop script cannot resolve OM nodes in HA

### PR DESCRIPTION
**Pull request by MLikeWater**
## What changes were proposed in this pull request?
Fix the problem that the Ozone start and stop script resolve OM node, and ensure that the start-ozone.sh and stop-ozone.sh command is used to manage the start and stop services of the Ozone cluster.

## What is the link to the Apache JIRA

[HDDS-6453](https://issues.apache.org/jira/browse/HDDS-6453)

## How was this patch tested?
1. configure password-free login between Ozone cluster nodes
2. start the Ozone service on an Ozone node
```
start-ozone.sh
```
3. stop the Ozone service on an Ozone node
```
stop-ozone.sh
```

### CODE REVIEW

File: hadoop-ozone/dist/src/shell/ozone/start-ozone.sh [hunk 1]
```diff
@@ -82,7 +82,8 @@ OZONE_JUMBO_RETCOUNTER=$?
 
 #---------------------------------------------------------
 # Ozone ozonemanager nodes
-OM_NODES=$("${OZONE_HOME}/bin/ozone" getconf -ozonemanagers 2>/dev/null)
+TMP_OM_NODES=$("${OZONE_HOME}/bin/ozone" getconf -ozonemanagers 2>/dev/null)
+OM_NODES=`echo $TMP_OM_NODES | grep -o '\([0-9]\{1,3\}\.\)\{3\}[0-9]\{1,3\}' | sort -d | uniq`
```
Comment (kerneltime) [inline hadoop-ozone/dist/src/shell/ozone/start-ozone.sh:86]: Please add a comment for what the intent of this line is.

Comment (MLikeWater) [reply hadoop-ozone/dist/src/shell/ozone/start-ozone.sh:87]: Ok, thanks for the tip.

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Please see [my comment](https://issues.apache.org/jira/browse/HDDS-6453?focusedCommentId=17529876&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17529876) on the Jira.

**Comment by MLikeWater** [comment]
@adoroszlai Obtain OM nodes list by using the command `ozone getconf -ozonemanagers`.
Please review.

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/OmUtils.java [hunk 2]
```diff
@@ -113,9 +113,10 @@ public static InetSocketAddress getOmAddress(ConfigurationSource conf) {
    * @param conf {@link ConfigurationSource}
    * @return {service.id -> [{@link InetSocketAddress}]}
    */
-  public static Map<String, List<InetSocketAddress>> getOmHAAddressesById(
+  public static Collection<InetSocketAddress> getOmHAAddressesById(
```
Comment (adoroszlai) [inline hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/OmUtils.java:116]: We would like to add structured output later (item 3 from the Jira issue).  So please keep `getOmHAAddressesById()` (and `testGetOmHAAddressesById()`) unchanged, and add a new method that flattens the map to a list.

One way to implement that, but you can also use loops:

```java
  public static Collection<InetSocketAddress> getOmAddresses(
      ConfigurationSource conf) {
    return getOmHAAddressesById(conf).values().stream()
        .flatMap(Collection::stream)
        .collect(toList());
  }
```

This should be called from `OzoneManagersCommandHandler`, and a new test method in `TestOmUtils`.

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @MLikeWater for the patch and sorry for the delay in review.

`TestGetConfOptions` is [failing](https://github.com/apache/ozone/runs/6556312747?check_suite_focus=true#step:5:2260) and needs to be updated.

**Comment by kerneltime** [comment]
@MLikeWater do you plan to revisit this PR and address pending review comments?

**Pull request closed without merging**